### PR TITLE
Fixed @LoggingScope annotation bug

### PR DESCRIPTION
**Pull request by sjlbos**
- Fixed bug that caused LoggingScope annotations to be ignored when they were placed on read-only fields (private fields with no setter)
- Updated annotation tests to include more test cases

### CODE REVIEW

File: src/test/java/com/godaddy/logging/models/AnnotatedObject.java [hunk 1]
```diff
@@ -27,27 +27,43 @@
 import com.godaddy.logging.LoggingScope;
 import com.godaddy.logging.Scope;
 
+public class AnnotatedObject {
 
-import lombok.Data;
```
Comment (devshorts) [inline src/test/java/com/godaddy/logging/models/AnnotatedObject.java]: Why was lombok removed? Lombok is specifically used for testing here to make sure it works as its common used.  If we're worried about it not working with standard java beans, lets create another class that has that as well

**Comment by devshorts** [comment]
Does this address #16 ?

**Comment by devshorts** [comment]
@jakeswenson @bcam909 can you take a look as well?

**Comment by bcam909** [comment]
@devshorts Yeah this does address #16

File: src/test/java/com/godaddy/logging/models/AnnotatedObject.java (hunk 1 above)
Comment (bcam909) [inline src/test/java/com/godaddy/logging/models/AnnotatedObject.java]: Yes can we keep this class as is? We can create a new class along with a new test that tests the logging scope for a field that is read only.

Comment (sjlbos) [inline src/test/java/com/godaddy/logging/models/AnnotatedObject.java]: It gets difficult to read/write a class that intermixes lombok and traditional Java while still covering all necessary test cases; that was my reasoning behind removing Lombok initially. I've updated the PR to include a second test case for Lombok-annotated classes.

**Comment by bcam909** [comment]
Get @akropp to approve but this looks good to me.

**Pull request merged**